### ISSUE TITLE
Support for IPV6 addresses via Host field on Add Download Client

### ISSUE THREAD

**Issue by Ttfgggf** [Sonarr/Sonarr#7534]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Current Behavior

So if I add an IPv4 address or subdomain, it seems to work.
If I add an IPv6 address even with square brackets or without, it somehow cannot connect to my download client.

### Expected Behavior

IPv6 addresses (whether ULA or GUA) can be used within the host field for Add Download Client.

### Steps To Reproduce

1. Set up Sonarr as usual.
2. Add Download Client with IPv6 address and it will say it cannot connect to qBittorrent.

### Environment

```markdown
- OS: Fedora 41
- Sonarr: Sonarr 4.0.11.2680
- Docker Install: Yes
- Using Reverse Proxy: No
- Browser: Firefox 90
- Database: SQlite
```

### What branch are you running?

Main

### Trace Logs?

Here is a log of what happened
https://privatebin.net/?0c1ae5131ad8756a#72kGb62wkvgtaDC4RpUD8394E2bQLuRybcNsmXrUqboj

### Anything else?

![Screenshot 2024-12-31 170453](https://github.com/user-attachments/assets/c85b2940-41d8-490e-80f1-dc9b618e37b3)

**Comment by Ttfgggf** [comment]
Also I am facing similar issue to this raised issue (https://github.com/Prowlarr/Prowlarr/issues/2157) in regards to anything IPv6.
I have switched to a Global Unicast Addresses, but I seem to have DNS issue after issue in regards to the *arr stack.
I have manual mappings for my DNS on router, which I though would solve this but they haven't.

**Comment by Ttfgggf** [comment]
> According to [RFC2732](https://www.ietf.org/rfc/rfc2732.txt), you should use square brackets for the host like`[fd3a:9a29:b136:eeee:aaaa::6]`.

I have tried that already. Sonarr doesn't seem to like it.

**Comment by Ttfgggf** [comment]
> What doesn't seems to like it? Share errors, details etc.

I have shared a log above already. But let me share it again. (Please be aware I have omitted part of my IPv6 address from the logs provided, but you should be able to see the errors.

Here is a privatebin link (Although I have omitted the address and changed over to Global Unicast Address, it still the same problem)

https://privatebin.net/?f64bc135b94ce504#GJfXaqMTKnTTMTp1WYhyhF2DNTjy8aKN7Uq7ctEJXiMc

![Screenshot 2025-01-02 123157](https://github.com/user-attachments/assets/60901e65-7576-4460-a706-1bfe75859924)

**Comment by Ttfgggf** [comment]
> > I have shared a log above already. But let me share it again. (Please be aware I have omitted part of my IPv6 address from the logs provided, but you should be able to see the errors.
> 
> 
> 
> Where, original message? Without brackets it's just an Uri parsing error so just after you fixed the host you triggered the error in HttpClient. In the end, very different errors.
> 
> 
> 
> 
> 
> 
> 
> @markus101 FYI regression from 682d2b4e1b8acb74ab6e75b85e2a73aa431b5332, no more tooltips in `FormInputGroup` with `DetailedDescription`. 😭 

This is the one.

https://privatebin.net/?0c1ae5131ad8756a#72kGb62wkvgtaDC4RpUD8394E2bQLuRybcNsmXrUqboj

Although, I changed from a ULA to a GUA as you can see in the above picture. I did to see if it was going to change anything but it didn't.

What additional things do I need to grab to add to this issue if needed?

**Comment by markus101** [comment]
The socket connection to qbit via IPv6 fails, is that some side effect of IPv6 being disabled within Sonarr?

**Comment by Ttfgggf** [comment]
> The socket connection to qbit via IPv6 fails, is that some side effect of IPv6 being disabled within Sonarr?

Could be. 
I believe Sonarr prefers IPv4 to IPv6 even if there are both addresses (Feel free to correct me on this)

**Comment by markus101** [comment]
Since a proper Happy Eyeballs implementation release is unknown it is probably worth moving from our completely disable IPv6 approach to something that includes caching for different hostnames or something closer to Happy Eyeballs, such as https://slugcat.systems/post/24-06-16-ipv6-is-hard-happy-eyeballs-dotnet-httpclient/

In both approaches caching to avoid retrying IPv6 and IPv4 in all cases is probably advisable.

**Comment by schmensch** [comment]
Hi, I just tested this again in Sonarr, Radarr and Prowlarr, and while Radarr and Prowlarr correctly accept addresses such as `[::1]` for the download client, and even prefer AAAA records if available (e. g. using `::1` first when the client is at `localhost`), Sonarr still fails or just uses the A records.

Is this related to the upstream dotnet bug, since Sonarr is still on Dotnet 6 while Radarr and Prowlarr have moved on to dotnet 8?

**Comment by markus101** [comment]
If things are working correctly in Radarr then they should be working in Sonarr once v5 is ready for beta/release.

**Comment by schmensch** [comment]
Yes, I'll test again with the latest `v5-develop` commit once I have time.

**Comment by Napsterbater** [comment]
> Hi, I just tested this again in Sonarr, Radarr and Prowlarr, and while Radarr and Prowlarr correctly accept addresses such as `[::1]` for the download client, and even prefer AAAA records if available (e. g. using `::1` first when the client is at `localhost`), Sonarr still fails or just uses the A records.
> 
> Is this related to the upstream dotnet bug, since Sonarr is still on Dotnet 6 while Radarr and Prowlarr have moved on to dotnet 8?

I was coming here (and to Radarr) to open a Bug for this and saw this one and your comment.

On Windows with Sonarr I of course have the issue and see it might be fixed in v5, but I also see the issues in Radarr, both act identically.

I get the message 

"Test was aborted due to an error: Unable to connect to SABnzbd, The requested name is valid, but no data of the requested type was found. ([::1]:11014)"

Now this is a SABnzbd API connection and not a qBittorrent connection so that might make a difference.

Devs, Should I open a separate Issue for this as its with a SABnzbd connection vs qBittorrent?

**Comment by markus101** [comment]
No, this issue is about download clients in general.

**Comment by Napsterbater** [comment]
> No, this issue is about download clients in general.

So then that goes back to, I see identical behavior in the current version of Radarr as Sonarr, which by your comments seemingly may mean it's not currently fixed in v5? Even though the other reporter said it worked for them in the current version of Radarr.

**Comment by schmensch** [comment]
> > Hi, I just tested this again in Sonarr, Radarr and Prowlarr, and while Radarr and Prowlarr correctly accept addresses such as `[::1]` for the download client, and even prefer AAAA records if available (e. g. using `::1` first when the client is at `localhost`), Sonarr still fails or just uses the A records.
> > Is this related to the upstream dotnet bug, since Sonarr is still on Dotnet 6 while Radarr and Prowlarr have moved on to dotnet 8?
> 
> I was coming here (and to Radarr) to open a Bug for this and saw this one and your comment.
> 
> On Windows with Sonarr I of course have the issue and see it might be fixed in v5, but I also see the issues in Radarr, both act identically.
> 
> I get the message
> 
> "Test was aborted due to an error: Unable to connect to SABnzbd, The requested name is valid, but no data of the requested type was found. ([::1]:11014)"
> 
> Now this is a SABnzbd API connection and not a qBittorrent connection so that might make a difference.
> 
> Devs, Should I open a separate Issue for this as its with a SABnzbd connection vs qBittorrent?

Not a developer of Sonarr or SABnzbd, but is this confirmed to be a bug related to the *arrs and IPv6? Does it work using the IPv4 address (127.0.0.1) instead? Does SABnzbd bind to IPv6 (:: or ::1) and does it work with other tools over IPv6?
